src/transformers/models/llama/modeling_llama.py

```diff
@@ -538,7 +640,7 @@ def _flash_attention_forward(
                 max_seqlen_k=max_seqlen_in_batch_k,
                 dropout_p=dropout,
                 softmax_scale=softmax_scale,
-                causal=True,
+                causal=self.attention_mask_cache.is_causal,
```

Comment:
This allows us to easily copy-paste this function to non-causal attention layers (BERT)